src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -127,17 +127,71 @@ public function getMaxDepth(): ?int
     /**
      * {@inheritdoc}
      */
-    public function setSerializedName(string $serializedName = null)
+    public function setSerializedNames(array $serializedNames): void
     {
-        $this->serializedName = $serializedName;
+        $this->serializedNames = $serializedNames;
+    }
+
+    /**
+     * Set a serialization name for given groups.
+     *
+     * @param string[] $groups
+     */
+    public function setSerializedName(string $serializedName = null/* , array $groups = [] */)
+    {
+        if (\func_num_args() < 2) {
+            trigger_deprecation('symfony/serializer', '6.2', 'The "%s()" method will have a new "array $groups = []" argument in version 7.0, not defining it is deprecated.', __METHOD__);
+            $groups = [];
+        } else {
+            $groups = func_get_arg(1);
+
+            if (!\is_array($groups)) {
+                throw new \TypeError(sprintf('"%s": Argument $groups is expected to be an array, got "%s".', __METHOD__, get_debug_type($groups)));
```

Comment:
the message must match what PHP natively would trigger:
`throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be array, "%s" given.', __METHOD__, get_debug_type($groups)));`